src/awkward/_backends.py

```diff
@@ -163,6 +167,34 @@ def __init__(self):
     def __getitem__(self, index: KernelKeyType) -> TypeTracerKernel:
         return TypeTracerKernel(index)
 
+    def _coerce_ufunc_argument(self, x):
+        if isinstance(x, ak._typetracer.TypeTracerArray):
+            return numpy.empty((0,) + x.shape[1:], dtype=x.dtype)
+        # Convert scalars to 0-d arrays
+        elif isinstance(x, ak._typetracer.UnknownScalar):
+            return numpy.empty((0,), dtype=x.dtype)
+        elif x is ak._typetracer.UnknownLength:
+            return numpy.empty((0,), dtype=np.int64)
+        elif isinstance(x, ak._typetracer.MaybeNone):
+            return self._coerce_ufunc_argument(x.content)
```

Comment:
@jpivarski this is one part I'm hesitant on. We're throwing away type information here because this *might* not fail at runtime, i.e. if the value is not actually none.

Reply:
If we *do* want this, should `MaybeNone` also support operators, e.g.. `MaybeNone(x) + y`? I'm inclined to say yes - `MaybeNone` should be fairly transparent, like Haskell's `Maybe`.

Reply:
And, what should we do about `OneOf`? The *proper* approach would be to try all branches and re-wrap the result, but if one branch fails, we would need to handle that.

Reply:
Our strategy is for typetracers to give code the benefit of the doubt and let it fail at runtime. The typetracer is not a type-check.

So if a typetracer (including `MaybeNone` or `OneOf`) represents $x$ or $y$ and $f(x)$ would yield some result but $f(y)$ would raise an error—great! We say that the result is $f(x)$.

If a typetracer represents $x$ or $y$ and both $f(x)$ and $f(y)$ would yield results, and they're different, that's a harder situation to deal with. I hope we don't have to because there might not be a good solution. We're not prepared to propagate non-deterministic type information. (If we had to, maybe we could go with the information-losing process of finding a common superclass of both results, like `np.numerical`.)

Reply:
OK, that sounds like the best path to me, too. Let's go with the safe, minimal approach and see what happens. I haven't implemented `OneOf` here, but we can do that in another PR, as it's probably better to keep momentum of the changes that are already in the pipeline.